File: src/main/java/kitchenpos/product/tobe/domain/Product.java
package kitchenpos.product.tobe.domain;

import jakarta.persistence.*;
import kitchenpos.common.tobe.Profanities;

import java.util.UUID;

@Table(name = "product")
@Entity
public class Product {
    @Column(name = "id", columnDefinition = "binary(16)")
    @Id
    private UUID id;

    @Embedded
    private ProductName name;

    @Embedded
    private ProductPrice price;

    protected Product() {}

    public Product(String name, long price, Profanities profanities) {
        this.id = UUID.randomUUID();
        this.name = new ProductName(name, profanities);
        this.price = new ProductPrice(price);
    }

    public UUID getId() {
        return id;
    }

    public ProductName getName() {
        return name;
    }


    public ProductPrice getProductPrice() {
        return price;
    }
    public Long getPrice() {
        return price.getPrice();
    }

    public void updatePrice(final ProductPrice productPrice) {
        this.price = productPrice;
    }
}

File: src/main/kotlin/kitchenpos/product/tobe/domain/Product.kt
package kitchenpos.product.tobe.domain

import jakarta.persistence.Column
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.*
import org.springframework.data.domain.AbstractAggregateRoot

@Table(name = "product")
@Entity(name = "TobeProduct")
class Product(
    @Column(name = "id", columnDefinition = "binary(16)")
    @Id
    var id: UUID = UUID.randomUUID(),

    @Embedded
    val productName: ProductName,

    @Embedded
    var productPrice: ProductPrice
) : AbstractAggregateRoot<Product>() {
    fun changePrice(productPrice: ProductPrice) {
        this.productPrice = productPrice
        registerEvent(ProductPriceChangedEvent(id))
    }
}
